cub/cub/detail/fast_modulo_division.cuh

```diff
@@ -109,11 +109,14 @@ multiply_extract_higher_bits(T value, R multiplier)
 {
   static_assert(supported_integral<T>::value, "unsupported type");
   static_assert(supported_integral<R>::value, "unsupported type");
-  _CCCL_DIAG_PUSH
-  _CCCL_DIAG_SUPPRESS_ICC(186) // pointless comparison of unsigned integer with zero
-  _CCCL_ASSERT(value >= 0, "value must be non-negative");
-  _CCCL_ASSERT(multiplier >= 0, "multiplier must be non-negative");
-  _CCCL_DIAG_POP
+  _CCCL_IF_CONSTEXPR (_CCCL_TRAIT(is_signed, T))
+  {
+    _CCCL_DIAG_PUSH
+    _CCCL_DIAG_SUPPRESS_ICC(186) // pointless comparison of unsigned integer with zero
+    _CCCL_ASSERT(value >= 0, "value must be non-negative");
+    _CCCL_ASSERT(multiplier >= 0, "multiplier must be non-negative");
+    _CCCL_DIAG_POP
+  }
```

Comment:
```suggestion
  _CCCL_ASSERT(_CUDA_VSTD::cmp_greater_equal(value, 0), "value must be non-negative");
  _CCCL_ASSERT(_CUDA_VSTD::cmp_greater_equal(multiplier, 0), "multiplier must be non-negative");
```
Once #2805 is merged, I think applying integer comparison functions should fix the warning.

Reply:
It would fix the warning but it would come at a cost.

We do not want a function call if we already know that the condition is statically satisfied